src/processing/ParameterizedOperationsUtil.java

```diff
@@ -63,6 +104,63 @@ public static BoardObject colorChangeUtil(BoardObject obj, UserId id, Intensity
         return newObj;
     }
 
+    /**
+    * Function to rotate the object, store all required object's data members in local
+    * variables, call for removing previous object from all utility maps, and finally create
+    * new updated object.
+    * To be called by ClientSideProcessing(via rotation function) and UndoRedo sections.
+    */
+    public static BoardObject rotationUtil(BoardObject obj, UserId id, Angle angleOfRotation) {
+
+        // storing initial list of pixels (to be used in creation of new BoardObject)
+        ArrayList<Pixel> prevPixelIntensities = new ArrayList<Pixel>();
+        prevPixelIntensities = obj.getPixels();
+
+        // finding centre of rotation (i.e centre Position of previous pixels)
+        Position centre = new Position(findCentre(prevPixelIntensities));
+
+        // Calculating the required rotation matrix
+        double[][] rotMatrix = rotationMatrix(angleOfRotation);
+
+        // generating new list of object's pixels
+        ArrayList<Pixel> newPixelSet = new ArrayList<Pixel>();
+        newPixelSet = obj.getPixels();
+        for(int i = 0; i < newPixelSet.size(); i++) {
+
+            // shifting origin
+            Position posWithShiftedOrigin = new Position(newPixelSet.get(i).position.x - centre.x,
+            newPixelSet.get(i).position.y - centre.y);
+
+            // applying rotation matrix followed by Narrowing Type Casting
+            Position rotatedPos = new Position((int)(rotMatrix[0][0] * posWithShiftedOrigin.x +
```

Comment:
This is a correct but huge expression. Please make it cleaner by assigning `posWithShiftedOrigin.x` and `posWithShiftedOrigin..y`to local variables and then multiplying with `rotMatrix`array values.

Reply:
but the number of terms in the expression remains same. And adding new variables will not be neat I guess.

Reply:
yes number of terms remains the same, but the size of the expression will reduce making it more readable.